### PR TITLE
Feature/cleaner treeview

### PR DESCRIPTION
**Pull request by ayurmedia**
Improve treeview with hidden rows and fields, and show some information on hover only.

### CODE REVIEW

File: Cleaner Treeview/CleanerTreeview.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+tau.mashups
+    .addDependency('tp3/mashups/topmenu')
+    .addDependency('tau/core/bus.reg')
```
[the remark this answers is not in the document]
Comment (ayurmedia) [reply Cleaner Treeview/CleanerTreeview.js]: i remove it, was from old mashup example copied over.

**Comment by ayurmedia** [comment]
> Thank you, that's cool mashup for some cases. May be just rethink a couple of potential issues:
> 
> 1. Now add and lookup buttons are hidden. Can't find 'add...' menu when clicking on item, and have no menu for lookup button, so need to always open card to add children.
> 2. Table headers for cells and hierarchy levels aren't always match vertically (need to setup list in a such way), so it is not clear for me which cell attribute i'm looking at (see feature `Teams` and story `Sprint` columns in examples), even when i hover mouse over cells / hierarchy levels.
> 
> Is it all intended?

Generally works fine for me. 
I can still add Sub-Items, eg. add a "Task" below a "User Story" when right-click over "User-Story". 
On mouse-hover all fields are visible and editable in that row and some context around. Its a bit difficult to find the perfect CSS selector. 
If you move the mouse outside the table, eg. to the top menu then you get the clean look. If you move the mouse inside the table, you see all details and can edit it. It's a compromise between clarity and details. 

In the Top Menu is now the "Tree" Menu-Link, with that you can toggle the compact-view on and off. 
When it is off, you have original functionality like without the mashup. When you toggle it on, you get a peaceful simplified view just reading a Bullet-List with a bit of Completed-State information. 

I also use it in combination with the Colorful User-Contributed Mashup (where i changed the font-size back to original) which helps a lot visually to see what is feature, user-story, task etc.  

Like the comment hints in the last line of JavaScript one could make the mashup without auto load, and keep the original behavior and press the "Tree" Top-Menu Link to turn the "focused mode" on. But in my workflow i prefer to have the simple view to start and then get into details when I want to edit. 

Adding Sub-Items is also quick and easy if you click on a row, then the overlay of the Ticket opens and you can easily add sub-tickets. There the tree view is flat and you still have the (+) button for quickly adding new ones. 

I don't see usability problems. 

It's open for discussion if the hover-effect is too nervous. One alternative could be to remove the hover- effect and stay in compact mode and do a hard mode switch with the top menu "tree" link if one wants more details and edit-features. For me the hover-effect is fine and I feel its efficient, but I might optimize the CSS over time when I use it for long time. As this is the official Mashup Repo I want to follow your suggestion so its good UX for most users. For my workflow it is usable in the current state.

File: Cleaner Treeview/README.mkd [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+The (+) Button is now hidden, but this functionality is easy to reach with a right-click "add ..." from the context menu.
+Or alternatively click on the parent Ticket and then Edit the Sub-Tickets as usual, also with the (+) Button.
+
+The 1st Table-Head Row is always Visible, which is useful in the sub-pages and still having enough navigation there.
+
+the column for "status" is still visible, because in my case thats important information i dont want to hide.
+
+By changing the CSS one can show and hide specific columns, if you find the css selector for the column.
+
+Before, Very noisy list lot of redundant information
```
[the remark this answers is not in the document]
Comment (ayurmedia) [reply Cleaner Treeview/README.mkd]: fixed typo

**Comment by tsarevichdm** [comment]
Good points, so fix small code issues and we'll merge it.

File: Cleaner Treeview/CleanerTreeview.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            style.appendChild(document.createTextNode(""));
+            document.head.appendChild(style);
+            var sheet = style.sheet;
+            var addCSSRule = function(selector, rules, index) { if (sheet.insertRule) { sheet.insertRule(selector + "{" + rules + "}", index); } else { sheet.addRule(selector, rules, index); }};
+            
+            // Modify Opacity
+            addCSSRule(".tau-list-caption > .tau-elems-table:not(.tau-elems-table-level-0)"," opacity:0; height:1px; ");
+            addCSSRule(".i-role-cardsholder:not(:hover) :not(.tau-list-time) > .tau-list-line > .tau-elems-table  .tau-elems-cell:not(.tau-elems-cell--details-trigger):not(:nth-child(1)):not(:nth-child(2)):not(.tau-list-state_full_length-cell) ","opacity: 0;");
+         };
+
```
[the remark this answers is not in the document]
Comment (ayurmedia) [reply Cleaner Treeview/CleanerTreeview.js]: done

File: Cleaner Treeview/README.mkd [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+After, simplified View, more like a normal Tree-List, much less redundant labels
+![After](https://github.com/TargetProcess/TP3MashupLibrary/raw/master/Cleaner%20Treeview/CleanerTreeview-2.png)
+
+
+How to activate it:
+
+1. Install the mashup from the Targetprocess Mashups Library
+2. Automatically TreeViews will have some hidden fields, which show up again on hover.
+3. If you miss too many fields, just deaktivate the mashup.
```
[the remark this answers is not in the document]
Comment (ayurmedia) [reply Cleaner Treeview/README.mkd]: Fixed typo, improved instructions.

File: Cleaner Treeview/CleanerTreeview.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ayurmedia) [reply Cleaner Treeview/CleanerTreeview.js]: done

File: Cleaner Treeview/CleanerTreeview.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ayurmedia) [reply Cleaner Treeview/CleanerTreeview.js]: done

**Comment by ayurmedia** [comment]
> 2\. so it is not clear for me which cell attribute i'm looking at

In my opinion the values are enough to understand if it's a team name or a release label. 
If its really difficult then the user has to turn the focus-mode off and gets all headers back, with the help of the top menu "Tree" toggle link. 
On the other hand it raises a UX question why two different field types even show up in the same column. Maybe better UX to keep it strict and move "release" and "teams" to separate columns, and have some empty spaces. Even with visible headers this might be confusing at the moment.

**Comment by tsarevichdm** [comment]
Cool, thank you!

**Pull request merged**